### ISSUE TITLE
[LINT_BUG]:  Objects exported from packages and functions run in purrr::map causes lint errors

### ISSUE THREAD

**Issue by vibalre** [Appsilon/box.linters#124]
### box.linters version

0.10.0

### Sample source code to lint

```r
# module.R
#' @export
my_constant <- 42

#' @export
my_function <- function(x) {
  x >= 1
}

# object.R
box::use(
  dplyr[storms],
  purrr[map, map_chr],
  stringr[str_to_upper],
)

box::use(
  . / module[my_constant, my_function],
)

#' @export
my_names <-
  map_chr(
    names(storms),
    str_to_upper
  )

#' @export
my_new_constant <- my_constant / 3

#' @export
my_answer <- map(c(1, 2, 3), my_function)

```

### Lint command used

lintr::lint(
    "object.R",
    linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters)
)

### Lint result

```
Line 2[box_pkg_fun_exists_linter] Function not exported by package.
Line 2[box_unused_attached_pkg_fun_linter] Imported function unused.
Line 4[box_unused_attached_pkg_fun_linter] Imported function unused.
```

### Expected result

There should be no lint errors.

**Comment by radbasa** [comment]
@jakubnowicki I think I know what's going on here. Or rather, what's not going on.

A data object from a package is being imported. We currently use `getNamespaceExports()` to determine what functions a package exports. This does not return data objects.

And because the unused attached package function linter assumes all the attached objects from a package are function, it is complaining that `storm()` is not used. 

The unused `str_to_upper` is the same. The linter is looking for a function token with the name `str_to_upper`. The `str_to_upper` in the `map_chr()` call is a plain object token.

Issue 1 will need to wait for the full support for box namespace exports from `{box}`.

Issue 2 and 3 can be handled within the linter functions.

**Comment by radbasa** [comment]
Issues 2 and 3 handled by PR 128.

Leaving this issue open for issue 1.
